src/client/hud.cpp

```diff
@@ -807,10 +807,11 @@ void drawItemStack(
 		driver->draw2DRectangle(color, progressrect2, clip);
 	}
 
-	if (font != NULL && item.count >= 2) {
+	const std::string &count_text = item.metadata.getString("count", 0);
+	if (font && (item.count >= 2 || !count_text.empty())) {
 		// Get the item count as a string
-		std::string text = itos(item.count);
-		v2u32 dim = font->getDimension(utf8_to_wide(text).c_str());
+		std::string text = count_text.empty() ? itos(item.count) : count_text;
```

Comment:
If an itemstack with `count > 1` contains a non-empty `count` meta field, the stack count is replaced by the custom text in `count`. But when `count` is set to `""`, the previous non-empty value of `count` is still displayed, and nothing else happens. There's no way to revert to the stack count string at the moment.

Looking at the code, I suspect the ` || !count_text.empty()` in L811 prevents the L813 from running, which would otherwise detect that the `count` field is empty and revert to displaying the stack count.